src/extension.ts

```diff
@@ -123,9 +115,9 @@ async function openTelemetryOptInDialogIfNeeded(context: ExtensionContext) {
   const selection = await window.showInformationMessage(message, 'Accept', 'Deny');
   if (!selection) {
     //close was chosen. Ask next time.
+    setTimeout(openTelemetryOptInDialogIfNeeded, RETRY_OPTIN_DELAY_IN_MS, context);
```

Comment:
no, this doesn't work if the popup disappeared after 15 secs (that was my 1st try ;-)). You need to set the timer before showing the popup

Reply:
Ahh!, I though disappear == close, but not the case. It is still visible when you click the right bottom 🔔 . 

@fbricon How about the new logic?

Reply:
that works